002_simulation_structure/002_simulation_structure.tex

```diff
@@ -100,6 +100,210 @@ \subsection{Infrastructure}
+ \hline
+ \texttt{RequestMessage} & \texttt{ResponseMessage} & \texttt{Reply}: Returns \texttt{BoolResponse} \newline \texttt{Request}: Returns the value of a request \textit{e.g. a RequestLeaveFood message that requests an agent to leave 10 food would return 10 on \texttt{Request}} & Asks how much food is on the platform when it arrives at the (recipient) agent \\ 
+ \hline
+ \texttt{ResponseMessage} & N/A & N/A & Asks how much food an agent is planning to take \\ 
+ \hline
+\end{tabular}
+\end{center}
+From these basic categories 11 specific messages were developed that specified what was being asked, stated, requested or responded to.
+\begin{center}
+\begin{longtable}{p{4cm}p{2.5cm}p{4cm}p{4cm}}
+ \hline
+ \textbf{Message Type} & \textbf{Category} & \textbf{Description} & \textbf{Reply Type} \\ [0.5ex] 
+ \hline\hline
+ \texttt{AskFoodTaken} & \texttt{AskMessage} & Asks how much food an agent has (already) taken
+ & \texttt{StateFoodTaken} \\ 
+ \hline
+ \texttt{AskHP} & \texttt{AskMessage} & Asks how much HP an agent has & \texttt{StateHP} \\ 
+  \hline
+ \texttt{AskFoodOnPlatform} & \texttt{AskMessage} & Asks how much food is on the platform when it arrives at the (recipient) agent & \texttt{StateFoodOnPlatform} \\ 
+ \hline
+ \texttt{AskIntendedFoodIntake} & \texttt{AskMessage} & Asks how much food an agent is planning to take & \texttt{StateIntendedFoodIntake} \\ 
+ \hline
+ \texttt{StateFoodTaken} & \texttt{StateMessage} & States how much food an agent has taken &   \\
+ \hline
+ \texttt{StateHP} & \texttt{StateMessage} & States how much HP an agent has &   \\
+ \hline
+ \texttt{StateFoodOnPlatform} & \texttt{StateMessage} & States how food is on the platform when it arrives to the agent &  
+ \\
+ \hline
+ \texttt{StateIntendedFoodIntake} & \texttt{StateMessage} & States how food the agent is planning to take &  
+ \\
+ \hline
+ \texttt{RequestLeaveFood} & \texttt{RequestMessage} & Requests that an agent (presumably above) you leaves a certain amount of food on the platform & \texttt{BoolResponse}
+ \\
+ \hline
+ \texttt{RequestTakeFood} & \texttt{RequestMessage} & Requests that an agent (presumably above) you takes a certain amount of food on the platform & \texttt{BoolResponse}
+ \\
+ \hline
+ \texttt{BoolResponse} & \texttt{ResponseMessage} & Message affirming or rejecting a request &  
+\end{longtable}
+\end{center}
+The distinction between ``receiving" and ``reacting" becomes important. Because we wanted to ensure that \textit{receiving} any of these message types did not mean that the recipient had to \textit{react}, the ``reaction" was separated into an external function which the agent could optionally call after they extracted their message from the inbox. \newline
+The design works such that if Alice on Floor 11 wanted to ask Floor 12 to leave 10 food on the platform for her:
+\begin{enumerate}
+    \item At Tick 0, Alice constructs a \texttt{RequestLeaveFoodMessage} with \texttt{Request} set to 10 addressed to Floor 12.
```

Comment:
```suggestion
The design works such that if Alice on Floor 11 wanted to ask Floor 10 to leave 10 food on the platform for her:
\begin{enumerate}
    \item At Tick 0, Alice constructs a \texttt{RequestLeaveFoodMessage} with \texttt{Request} set to 10 addressed to Floor 10.
```

Tower is inverted, its all so confusing. But sending upwards is done to a lower valued floor